File: HACKERRANK/Problem Solving/Algorithms/Implementation/Birthday Chocolate/bdaychoco.py
import sys
length, count = int(input()), 0

choco_arr = [int(x) for x in input().split()]
day, month = (int(x) for x in input().split())
if length == 1:
    print(1)
    sys.exit(0)
i = 0
while month <= length:
    if sum(choco_arr[i:month]) == day:
        count += 1
    i += 1
    month += 1

print(count)

File: HACKERRANK/Problem Solving/Algorithms/Implementation/Birthday Chocolate/bdaychoco.go
package main

import "fmt"

func sum(arr []int) int {
	val := 0
	for _, elem := range arr {
		val += elem
	}
	return val
}

// m squares summed up to d
func count(bar []int, n int, m int, d int) int {
	count := 0
	for i := 0; i <= n-m; i++ {
		if sum(bar[i:i+m]) == d {
			count += 1
		}
	}
	return count
}

func read_arr(n int) ([]int, error) {
	arr := make([]int, n)
	for i := range arr {
		_, err := fmt.Scan(&arr[i])
		if err != nil {
			return arr[:i], err
		}
	}
	return arr, nil
}

func main() {
	//Enter your code here. Read input from STDIN. Print output to STDOUT
	var n, d, m int
	fmt.Scanf("%v", &n)
	bar, _ := read_arr(n)
	fmt.Scanf("%v", &d)
	fmt.Scanf("%v", &m)
	fmt.Println(count(bar, n, m, d))
}
